smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -128,20 +131,67 @@ contract MixinTransfer is
     public
   {
     _isValidKey(_tokenId);
-    if(
-      // the specified address does not own the token
-      ownerOf(_tokenId) != _from
-      || 
-      ( // the sender is not owner or authorized
-        ownerOf(_tokenId) != msg.sender
-        && approved[_tokenId] != msg.sender
-        && !isApprovedForAll(_ownerOf[_tokenId], msg.sender)
-      )
-      || // a key manager is set
-      keyManagerOf[_tokenId] != address(0)
-    ) {
+    _onlyKeyManagerOrApproved(_tokenId);
+    
+    // revert if key manager is set
+    if(keyManagerOf[_tokenId] != address(0)) {
+      revert UNAUTHORIZED();
+    }
+
+    _transferFrom(_from, _recipient, _tokenId);
+  }
+
+  /** 
+  * Lending a key allows you to transfer the token while retaining the 
+  * ownerships right by setting yourself as a key manager first
+  * @param _from the owner of token to transfer
+  * @param _recipient the address that will receive the token
+  * @param _tokenId the id of the token
+  * @notice Only the key owner or the key manager can call this function. If the owner calls it and no
+  * key manager is set, then the owner will be set as key manager.
+  */
+
+  function lendKey(
+    address _from,
+    address _recipient,
+    uint _tokenId
+  )
+    public
+  {
+    // make sure caller is either owner or key manager 
+    if(msg.sender != keyManagerOf[_tokenId] && msg.sender != ownerOf(_tokenId)) {
```

Comment:
What happens if the key manager is not set? Does it mean the key owner has to explicitly set themselves as key manager first and then call lendKey?

Reply:
Ideally we would allow the lendKey if the caller is the ownerOf(_tokenId) AND keyManagerOf[_tokenId] is not set, and set the keyManagerOf _inside_ this function.

Reply:
```suggestion
    if(msg.sender != _isKeyManager(_tokenId)) {
```

Reply:
ok. It means the keyManager needs to be set to address zero before using `lendKey` which is fine